src/jarabe/frame/activitiestray.py

```diff
@@ -90,6 +100,53 @@ def __init__(self, home_activity, group):
         elif home_activity.props.launch_status == shell.Activity.LAUNCH_FAILED:
             self._on_failed_launch()
 
+    def drag_drop_cb(self, widget, context, x, y, time):
+        self._drag_save = False
+        context_targets = context.list_targets()
+        for target in context_targets:
+            if str(target) not in ('TIMESTAMP', 'TARGETS', 'MULTIPLE'):
+                widget.drag_get_data(context, target, time)
+        return True
+
+    def journal_drag_data_received_cb(self, widget, context, x, y, selection,
+                              targetType, time):
+        if (hasattr(self,'_drag_save') and self._drag_save == True) or not hasattr(self,'_drag_save'):
+            return
+        logging.debug('Saved to Journal')
+        self._drag_save = True
+        data = None
+        file_path = None
+        if selection.get_text() != None:
+	    data = selection.get_text()
+        if len(selection.get_uris()) != 0:
+            file_path = selection.get_uris()[0]
+        if selection.get_pixbuf() != None:
+            data = selection.get_pixbuf()
+
+        if data != None:
+            tmp_dir = os.path.join(env.get_profile_path(), 'data')
+            f, file_path = tempfile.mkstemp(dir=tmp_dir)
+            try:
+                os.write(f, data)
+            finally:
+                os.close(f)
+        
+        if file_path != None:
+            jobject = datastore.create()
+            jobject.metadata['title'] = 'test'
```

Comment:
I wanted to do that but for it I needed to pass the clipboard object to the drag destination, but I was unable to do that, since I was only able to pass text, uris, etc but not all objects. Could you help me with this?